### PR TITLE
Variants Visualization

### PR DESCRIPTION
**Pull request by erictu**
Working visualization of variants when given vcf input to ADAM viz
Also added loadGenotypes and maybeLoadVcf in ADAMContext

### CODE REVIEW

**Comment by laserson** [comment]
Jenkins, retest this please.

**Comment by laserson** [comment]
@fnothaft We really need to fix that intermittent wigFix bug :-/

**Comment by fnothaft** [comment]
@laserson hah, yeah... I suppose we need to root-cause it first...

**Comment by laserson** [comment]
@erictu Could you rebase this on master?

File: adam-cli/src/main/scala/org/bdgenomics/adam/cli/VizReads.scala [hunk 1]
```diff
@@ -168,7 +210,14 @@ class VizReads(protected val args: VizReadsArgs) extends ADAMSparkCommand[VizRea
...
 
     val proj = Projection(contig, readMapped, readName, start, end)
-    VizReads.reads = sc.loadAlignments(args.inputPath, projection = Some(proj))
+
+    if (args.inputPath.endsWith(".bam") || args.inputPath.endsWith(".sam")) {
+      VizReads.reads = sc.loadAlignments(args.inputPath, projection = Some(proj))
+    }
+
+    if (args.inputPath.endsWith(".vcf")) {
+      VizReads.variants = sc.adamVCFLoad(args.inputPath).flatMap(_.genotypes)
```
Comment (fnothaft) [inline adam-cli/src/main/scala/org/bdgenomics/adam/cli/VizReads.scala]: Switch to `loadGenotypes`?

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/variation/VariationRDDFunctions.scala [hunk 2]
```diff
@@ -110,4 +110,5 @@ class GenotypeRDDFunctions(rdd: RDD[Genotype]) extends Serializable with Logging
         rec.getVariant.getEnd > query.start
     rdd.filter(overlapsQuery)
   }
-}
+
```
Comment (fnothaft) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/variation/VariationRDDFunctions.scala]: This is just a whitespace diff; can you change this back to the old version. It's preferable to not have space between the two braces, and to have the EOF after a newline.

File: adam-cli/src/main/webapp/WEB-INF/layouts/variants.ssp [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+        }
+
+        // Hover box for reads
+        var div = d3.select("body")
+                    .append("div")
+                    .attr("class", "tooltip")
+                    .style("opacity", 0);
+    </script>
+</body>
+</html>
```
Comment (fnothaft) [inline adam-cli/src/main/webapp/WEB-INF/layouts/variants.ssp]: Add a newline before EOF.

File: adam-cli/src/main/scala/org/bdgenomics/adam/cli/VizReads.scala [hunk 4]
```diff
@@ -168,7 +210,14 @@ class VizReads(protected val args: VizReadsArgs) extends ADAMSparkCommand[VizRea
     VizReads.refName = args.refName
 
     val proj = Projection(contig, readMapped, readName, start, end)
-    VizReads.reads = sc.loadAlignments(args.inputPath, projection = Some(proj))
+
+    if (args.inputPath.endsWith(".bam") || args.inputPath.endsWith(".sam")) {
```
Comment (fnothaft) [inline adam-cli/src/main/scala/org/bdgenomics/adam/cli/VizReads.scala]: Should also look for the ADAM extensions (#190 provides suggestions).

**Comment by fnothaft** [comment]
A few small nits, but otherwise LGTM!

**Comment by fnothaft** [comment]
Merged! Thanks @erictu!

**Pull request merged**